### PR TITLE
[CODEGEN][AOCL] Add math intrinsic rules

### PR DESCRIPTION
**Pull request by kazum**
This PR adds math intrinsic rules for AOCL.

In addition, to simplify an AOCL target name for emulation, I made the `-device` parameter of the AOCL target optional.

### CODE REVIEW

File: src/codegen/codegen_aocl.cc [hunk 1]
```diff
@@ -31,17 +31,16 @@ runtime::Module BuildAOCL(Array<LoweredFunc> funcs, std::string target_str) {
...
+  if (target->device_name != "") {
+    cmd += " -board=" + target->device_name;
   }
-  std::string cmd = "aoc aocl.cl";
   for (std::string option : target->options()) {
     if (option == "-mattr=emulator") {
       cmd += " -march=emulator";
     }
   }
-  cmd += " -board=" + target->device_name;
```
Comment (ktabata) [inline src/codegen/codegen_aocl.cc]: Removing target device name look good to me.
'aoc' uses 'AOCL_BOARD_PACKAGE_ROOT' environment variable to determine the board name when '-board=' is not specified.

**Comment by ktabata** [comment]
Removing target device name looks good to me.
'aoc' uses 'AOCL_BOARD_PACKAGE_ROOT' environment variable to determine the board name when '-board=' is not specified.

File: src/codegen/codegen_aocl.cc (hunk 1 above)
Comment (ktabata) [inline src/codegen/codegen_aocl.cc]: This works on my machine.

File: topi/tests/python/test_topi_math.py [hunk 2]
```diff
@@ -39,7 +39,8 @@ def check_device(device):
             foo(a, b)
             np.testing.assert_allclose(b.asnumpy(), b_np, rtol=1e-5, atol=1e-5)
 
-        for device in ['cuda', 'opencl', 'metal', 'rocm', 'vulkan', 'llvm', 'nvptx', 'sdaccel']:
+        for device in ['cuda', 'opencl', 'metal', 'rocm', 'vulkan', 'llvm', 'nvptx', 'sdaccel',
+                       'aocl -mattr=emulator']:
```
Comment (tmoreau89) [inline topi/tests/python/test_topi_math.py]: would it make sense to rename the device string to a something like aocl_emu? keeping in mind that there are different forms of emulation (sw vs. hw)

**Comment by tmoreau89** [comment]
Overall the changes look good. My only question touches on the way we pass the emulation attribute to the AOCL code gen backend. One way is to use a target option as we do, but another would be to simply add another target string like aocl_emu which makes it easier to parse targets strings.

**Comment by kazum** [comment]
@tmoreau89 Thanks for your comment.  Adding a target string for emulation sounds good to me.  I've pushed a commit for that, can you check it out?

**Comment by tmoreau89** [comment]
thank you, I think having an emulation target makes sense in this context. Does the aocl support different types of emulation (sw vs. hw as with sdaccel)?

**Comment by kazum** [comment]
@tmoreau89 AFAIK, AOCL supports only software emulation.  No need to care about different kind of emulation.

**Comment by nishi-t** [comment]
If we will support different emulation types for sdaccel by naming (e.g. `sdaccel_sw_emu` in the future, `sdaccel_hw_emu`), using consistent name(e.g. `aocl_sw_emu`) may be good for users. But this may be matter of taste :)

**Comment by tmoreau89** [comment]
indeed some folks might care about the distinction between hardware and software emulation, therefore it might be good to explicitly specify it since sdaccel offers both modes (to avoid any ambiguity). Since execution time can vary by 1-2 orders of magnitude from software to hardware emulation, it'll be good to have explicit control by changing the target name.

**Comment by tmoreau89** [comment]
LGTM

**Pull request merged**